src/validation.c

```diff
@@ -415,6 +420,51 @@ validate_sriov_rules(const NetplanParser* npp, NetplanNetDefinition* nd, GError*
     return valid;
 }
 
+gboolean
+adopt_and_validate_vrf_routes(const NetplanParser *npp, GHashTable *netdefs, GError **error)
+{
+    gpointer key, value;
+    GHashTableIter iter;
+
+    g_hash_table_iter_init (&iter, netdefs);
+    while (g_hash_table_iter_next (&iter, &key, &value))
+    {
+        NetplanNetDefinition *nd = value;
+        if (nd->type != NETPLAN_DEF_TYPE_VRF || !nd->routes)
+            continue;
+
+        /* Routes */
+        for (size_t i = 0; i < nd->routes->len; i++) {
+            NetplanIPRoute* r = g_array_index(nd->routes, NetplanIPRoute*, i);
+            if (r->table == nd->vrf_table)
+                continue;
+            else if (r->table != NETPLAN_ROUTE_TABLE_UNSPEC && r->table != nd->vrf_table) {
+                g_set_error(error, G_MARKUP_ERROR, G_MARKUP_ERROR_INVALID_CONTENT,
+                            "%s: VRF routes table mismatch (%d != %d)", nd->id, nd->vrf_table, r->table);
+                return FALSE;
+            } else {
+                r->table = nd->vrf_table;
```

Comment:
question: will the table be rendered out via `netplan get` ? Not sure, since it's not marked as dirty...

Reply:
It will not be rendered out via `netplan get`, which is what we want, as we want to rely on the VRF's `table` setting only. So I think that's all fine.

Reply:
Well, we've usually tried to write out the YAML as close to the original input, even if redundant, hence the dirtyness tracking. If the `table` field is defined in the route *and* in the vrf (and both match), we might want to output it in both places, or alternatively warn that it's redundant, perhaps in verbose mode?

Reply:
Thanks for the clarification, that makes a lot of sense.
I've added a debug log about ignoring the redundant `table` value in V3 of 362b85afdb6f8bd40737ca9989cccb1342bd10c2 to account for that for now.